src/dune_rules/melange/melange.ml

```diff
@@ -2,13 +2,13 @@ open Import
 
 module Module_system = struct
   type t =
-    | Es6
+    | ESM
     | CommonJs
 
   let default = CommonJs, ".js"
 
   let to_string = function
-    | Es6 -> "es6"
+    | ESM -> "es6"
```

Comment:
still printing to `"es6"` for the command-line option since that's what all Melange versions support.